### PR TITLE
Add a feature to pass arguments on to each task.

### LINKED ISSUE #41: Parameter/argument passthrough
Hi,

Can I pass an argument into an `npm-run-all` script from the command line?

I have defined these scripts:
````
"css": "grunt watch:css",
"mm": "bundle exec middleman",
"serve": "npm-run-all --parallel css mm",
````

I can pass an argument directly to the `mm` script, i.e. `npm run mm -- --port=4567`. Can I pass this argument in to the `serve` script from the command line and have it forwarded on to the `mm` script?

### PR DESCRIPTION
**Pull request by robario**
Closes #41

### CODE REVIEW

File: src/lib/index.js [hunk 1]
```diff
@@ -160,6 +162,22 @@ module.exports = function npmRunAll(
...
+        for (let i = 0, len = patterns.length; i < len; ++i) {
+            patterns[i] = patterns[i].replace(/[{]([*@]|\d+)[}]/g, (match, index) => {
+                if (index === "@") {
+                    return shellQuote.quote(rest.slice(1));
+                }
+                if (index === "*") {
+                    return shellQuote.quote([rest.slice(1).join(" ")]);
+                }
+                const position = parseInt(index, 10);
+                if (position >= 0 && position < rest.length) {
```
Comment (mysticatea) [inline src/lib/index.js]: Should is this `position >= 1`?

Comment (robario) [inline src/lib/index.js]: Not necessarily the case.
And say to C language, I think the `{0}` behaivior is **unspecified** but not **undefined**.

Comment (robario) [inline src/lib/index.js]: And in the current specification, `{0}` is always `--`.
Only a few, similar to a `$0` of shell script.

Comment (mysticatea) [inline src/lib/index.js]: Sounds good, thank you!

**Comment by robario** [comment]
Thank you for merging.
I expect official releasing it hourly.

**Pull request merged**